### DISCUSSION TITLE
Locale detection mismatch

### DISCUSSION THREAD

**Discussion by oskar-gmerek** [ivanhofer/typesafe-i18n#438] [Q&A] [answered]
### Version

5.12.0

### Describe the bug

I may be wrong and I think that may be more related to the browsers I also believe that in English it may works correctly as long English is an official language in the specific country. Anyway, I decided to share my feelings.

My app has 4 locations: en, en-GB, pl, pl-PL
My browsers are sending in headers: `accept-language: pl-PL,pl;q=0.9`.
typesafe-i18n location detector is redirecting me to /pl-pl, which looks good at first glance, but...

I'm physically in the GB, and my browser has settings to prefer the Polish Language, but I getting content in Polish for Poland `/pl-pl. 
I believe that I should get content for Polish Language `/pl` as long is available without a specific country.

For my use case, I will need to turn off language detections in typesafe-i18n and go for IP-detection-based redirections, but I think that behaviour should be changed anyway. 

### Reproduction

[typesafe-i18n-demo-sveltekit](https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit) + language with specific country code (depends on Browsers settings)

### Additional information

System: macOS 12.5.1 (Language: PL, location: GB)
Browser: Safari Browser (default configuration)
Browser: Brave Browser (default configuration)

SvelteKit: 1.0.0-next.476

**Comment by ivanhofer** [comment, marked as answer]
Hi @oskar-gmerek,
the example works as expected. The code in the example just uses the language-header for detection.
https://github.com/ivanhofer/typesafe-i18n-demo-sveltekit/blob/main/src/hooks.ts#L25-L28

If you want to use the IP-address as a detection strategy, you would need to write your own custom locaale detector. You can find more information in the [official docs](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/detectors#detectors-1)
